### PR TITLE
Migrate fullpage redirect to App Bridge CDN

### PR DESCRIPTION
**Pull request by kirillplatonov**
### What this PR does

Switching from hardcoded AppBridge 3 version to AppBridge CDN. This removes the need to pass `params[:host]` for `fullpage_redirect_to` helper and fixes redirect with token exchange auth. I manually tested it with both token exchange and legacy auth.

### Checklist

Before submitting the PR, please consider if any of the following are needed:

- [x] Update `CHANGELOG.md` if the changes would impact users
- [x] Update `README.md`, if appropriate.
- [x] Update any relevant pages in `/docs`, if necessary
- [x] For security fixes, the [Disclosure Policy](https://github.com/Shopify/shopify_app/blob/master/SECURITY.md#disclosure-policy) must be followed.

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-12.22.8
```
Comment (kirillplatonov) [inline .nvmrc:1]: Upgraded as v12 is no longer maintained and I couldn't install it locally

**Comment by kirillplatonov** [comment]
cc @zzooeeyy @paulomarg

**Comment by kirillplatonov** [comment]
@matteodepalo please take a look at this one too. It's important upgrade to move forward with migration to id_token.

**Comment by matteodepalo** [comment]
Hi @kirillplatonov, we're taking a look now.

**Review by lizkenyon — APPROVED** [review]

File: app/assets/javascripts/shopify_app/redirect.js [hunk 2]
```diff
@@ -10,9 +10,7 @@
 
     var targetInfo = JSON.parse(redirectTargetElement.dataset.target);
 
-    var appBridgeUtils = window['app-bridge']['utilities'];
-
-    if (appBridgeUtils.isShopifyEmbedded()) {
+    if (window['shopify']) {
```
[the remark this answers is not in the document]
Comment (kirillplatonov) [reply app/assets/javascripts/shopify_app/redirect.js]: Removed it as `open` should cover both embedded and non-embedded context now (great design decision)

File: test/javascripts/shopify_app/app_bridge_redirect_test.js [hunk 3]
```diff
@@ -1,66 +1,21 @@
...
-    const Redirect = AppBridge.actions.Redirect;
-    var mockApp = {};
-    var RedirectInstance = {
-      dispatch: sinon.spy(),
-    };
-    sinon.stub(AppBridge, 'default').callsFake(() => mockApp);
-    sinon.stub(Redirect, 'create').callsFake(() => RedirectInstance);
-
-    const normalizedUrl = `${window.location.origin}${url}`;
+  test('calls App Bridge redirect to normalized url', () => {
```
[the remark this answers is not in the document]
Comment (kirillplatonov) [reply test/javascripts/shopify_app/app_bridge_redirect_test.js]: Merged and renamed tests. To be honest, I don't like that kind of tests where everything is stabbed and no real behavior being tested. But not sure if there's a better way to test App Bridge. It's a common issue for testing app related code that works with App Bridge too.

**Comment by kirillplatonov** [comment]
@byrichardpowell made adjustments based on your feedback. Code become even simpler 👍

**Comment by netwire88** [comment]
Are there any instructions on how to use `fullpage_redirect_to`?

**Pull request merged**